utils/content.ts

```diff
@@ -1,21 +1,41 @@
 import fs from 'node:fs';
 import path from 'node:path';
 import matter from 'gray-matter';
-import { remark } from 'remark';
-import highlight from 'remark-highlight.js';
+import rehypeHighlight from 'rehype-highlight';
+import gfm from 'remark-gfm';
+import html from 'remark-html';
+import remarkParse from 'remark-parse';
+import { unified } from 'unified';
 import { v4 as uuid } from 'uuid';
+import { contentTypesMap } from './content-types';
 
 const workDirectory = path.join(process.cwd(), 'content', 'work');
 const notesDirectory = path.join(process.cwd(), 'content', 'notes');
 const articlesDirectory = path.join(process.cwd(), 'content', 'articles');
 
-type IContentType = 'articles' | 'notes' | 'work';
+export type IContent = {
+  title: string;
+  slug: string;
+  basePath: string;
+  date: Date;
+  id: string;
+  draft?: boolean;
+
+  selectedWork?: boolean;
+  description?: string;
+  previewImage?: string;
+
+  tags?: string[];
+  category?: string;
```

Comment:
Add `tags` and `category` to type `IContent` since these values can be optional and this type is used on multiple occasions.

Reply:
Should this be PRed to upstream vickonrails?